suite-common/trading/src/thunks/exchange/sendDexTransactionThunk.ts

```diff
@@ -0,0 +1,119 @@
+import { ExchangeTrade } from 'invity-api';
+
+import { createThunk } from '@suite-common/redux-utils';
+import { Account } from '@suite-common/wallet-types';
+
+import { exchangeThunks, tradingThunks } from '../';
+import { TRADING_EXCHANGE_THUNK_PREFIX } from '../../constants';
+import { tradingActions } from '../../reducers/tradingReducer';
+import { selectTradingExchangeSelectedQuote } from '../../selectors/tradingSelectors';
+import { RecomposeAndSignTxThunkProps } from '../common/recomposeAndSignTxThunk';
+
+export type SendDexTransactionThunkProps = {
+    account: Account;
+    returnUrl: string;
+    setMaxOutputId?: number | undefined;
+
+    nextStep: () => void;
+    processResponseData: (response: ExchangeTrade) => void;
+    triggerAnalyticsTradeConfirmation: () => void;
+    signAndPushSendFormTransaction: RecomposeAndSignTxThunkProps['signAndPushSendFormTransaction'];
+};
+
+export const sendDexTransactionThunk = createThunk(
+    `${TRADING_EXCHANGE_THUNK_PREFIX}/sendDexTransaction`,
+    async (
+        {
+            account,
+            returnUrl,
+            setMaxOutputId,
+            nextStep,
+            processResponseData,
+            triggerAnalyticsTradeConfirmation,
+            signAndPushSendFormTransaction,
+        }: SendDexTransactionThunkProps,
+        { dispatch, getState, rejectWithValue },
+    ) => {
+        const selectedQuote = selectTradingExchangeSelectedQuote(getState());
+
+        if (
+            !selectedQuote ||
+            !selectedQuote.dexTx ||
+            !selectedQuote.receiveAddress ||
+            (selectedQuote.status !== 'APPROVAL_REQ' && selectedQuote.status !== 'CONFIRM')
+        ) {
+            return rejectWithValue({
+                type: 'error',
+                error: { id: 'TR_TRADING_CANNOT_SEND_TRANSACTION' },
+            });
+        }
+
+        // after discussion with 1inch, adjust the gas limit by the factor of 1.25
+        // swap can use different swap paths when mining tx than when estimating tx
+        // the geth gas estimate may be too low
+        const { payload } = await dispatch(
+            tradingThunks.recomposeAndSignTxThunk({
+                account,
+                address: selectedQuote.dexTx.to,
+                amount: selectedQuote.dexTx.value,
+                destinationTag: selectedQuote.partnerPaymentExtraId,
+                ethereumDataHex: selectedQuote.dexTx.data,
+                recalculateCustomLimit: true,
+                ethereumAdjustGasLimit: selectedQuote.status === 'CONFIRM' ? '1.25' : undefined,
+                setMaxOutputId,
+                signAndPushSendFormTransaction,
+            }),
+        );
+
+        if (!payload || (payload && 'error' in payload) || !payload.success) {
```

Comment:
same nitpick as in the previous PR
```suggestion
        if (!payload || 'error' in payload || !payload.success) {
```

Reply:
Updated